packages/core/contracts/oracle/implementation/SafetyModule.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.16;
+
+import "./Finder.sol";
+import "./GovernorV2.sol"; //todo: swap this to using an interface.
+import "./Constants.sol";
+import "../interfaces/StakerInterface.sol";
+import "./AdminIdentifierLib.sol";
+import "../../common/implementation/Lockable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract SafteyModule is Ownable, Lockable {
+    using SafeERC20 for IERC20;
+    IERC20 public immutable token;
+    uint256 public bond;
+    GovernorV2 public immutable governor;
+    Finder public immutable finder;
+
+    struct EmergencyProposal {
+        address proposer;
+        uint64 time;
+        uint256 lockedBond;
+        uint256 cumulativeSignaled;
+        mapping(address => uint256) accountSignaled;
+        bool ratified;
+    }
+    EmergencyProposal[] public emergencyProposals;
+
+    uint256 emergencyActionThreshold = 0.5e18;
```

Comment:
Similar to the last implementation, a percentage cutoff means that during migrations, the system is more vulnerable and the multisig would have substantial power (i.e. collusion between multisig and another address could take over the system), right?

Reply:
correct.